.github/workflows/publish.yml

```diff
@@ -54,22 +56,58 @@ jobs:
         env:
           PACKAGES: ${{ github.event.inputs.packages }}
         run: |
+          if [[ "${{ github.event.inputs.packages }}" == "" ]]; then
+            packages="yew yew-macro yew-router yew-router-macro yew-agent yew-agent-macro" # publish all the crates
+          else
+            packages="${{ github.event.inputs.packages }}"
+          fi
+    
           output=""
-          for pkg in ${{ github.event.inputs.packages }}
+          for pkg in $packages
           do
             output+="--package $pkg " 
           done
           echo "pkg=$output" >> $GITHUB_ENV
 
+      - name: Inputs
+        shell: bash
+        run: |
+          if [[ "${{ github.event.inputs.level }}" == "" ]]; then
+            echo "level=minor" >> $GITHUB_ENV
```

Comment:
Does that mean, in 3 month, we will have 0.22, 0.23 and 0.24?

And every release can contain breaking changes?

That would result in a very fragmented ecosystem and a nightmare for library crate maintainers.

Reply:
Yes, but we should strive towards reducing breaking changes and having a more stable API

Reply:
However, as each crate must be targeting 1 minor release (pre 1.x), this forces every library maintainer to publish a release every month.

For an end user, they will not be able to update to a new version until all libraries releases a version that is compatible. And, for each release, currently, this can sometimes take weeks. (Some crates I used have yet to support Yew 0.21.)

> but we should strive towards reducing breaking changes and having a more stable API

If that is the case, we should consider a policy of no breaking changes for a certain period of time.
I do not think it would be reasonable for every user / library crate maintainer will be forced to migrate breaking change every couple months.

I think one of the primary reasons why rapid release cycle of Rust is so successful is because it is always backward compatible. Rust 1.73.0 will compile code written for Rust 1.0.0.

Reply:
Ah yes I thought it was for patch/minor releases for some reasons. Good point. Maybe we should have a branch per major version and make the automatic releases on the patch versions

Reply:
You're right. I did this automation for wasm-bindgen and copied the same approach for Yew. I didn't realize the difference that wasm-bindgen uses patch releases so nothing is broken between releases. I'll close this PR

Reply:
But don't we want to do it for patch releases? That would require us to maintain a branch per minor